R/tfpscanner.R

```diff
@@ -62,698 +70,796 @@ tfpscan <- function(tre,
+    gtr2 <- gtr1
+
+    if (length(allsegregating) < 100) {
+      gtr2 <- ggtree::gheatmap(gtr1,
+        as.data.frame(aas),
+        width = .66,
+        offset = 0.0005,
+        colnames = FALSE,
+        colnames_angle = -90,
+        colnames_position = "top",
+        colnames_offset_y = -2
+      ) +
+        ggplot2::theme(legend.position = "none")
+    }
+    gtr2
+  }
+
+  # all analyses for a particular node
+  .process.node <- function(u) {
+    tu <- descendantSids[[u]]
+    ta <- .get_comparator_sample(u)
+    ulins <- amd$lineage[match(tu, amd$sequence_name)]
+    alins <- amd$lineage[match(ta, amd$sequence_name)]
+    lgs <- .logistic_growth_stat(u, ta)
+    best_gr <- NA
+    if (!is.na(lgs$lgr)) {
+      best_gr <- lgs$growthrates[which.max(lgs$relative_model_support)]
+    }
+
+    reg_summary <- tryCatch(.region_summary(tu), error = function(e) as.character(e))
+    cocirc_summary <- tryCatch(.lineage_summary(ta), error = function(e) as.character(e))
+    lineage_summary <- tryCatch(.lineage_summary(tu), error = function(e) as.character(e))
+
+    a <- .get_comparator_ancestor(u)
+    if (compute_cluster_muts) {
+      cmut <- .cluster_muts(u, a)
+    } else {
+      cmut <- list(defining = NA, all = NA)
+    }
+
+    X <- data.frame(
+      cluster_id = as.character(u),
+      node_number = u,
+      parent_number = ifelse(is.null(ancestors[[u]]),
+        NA,
+        utils::tail(ancestors[[u]], 1)
+      ),
+      most_recent_tip = as.Date(
+        lubridate::date_decimal(max(stats::na.omit(sts[tu])))
+      ),
+      least_recent_tip = as.Date(
+        lubridate::date_decimal(min(stats::na.omit(sts[tu])))
+      ),
+      cluster_size = length(tu),
+      logistic_growth_rate = best_gr,
+      logistic_growth_rate_p = lgs$lgrp,
+      simple_logistic_growth_rate = lgs$lgr,
+      gam_logistic_growth_rate = lgs$gam_r,
+      simple_logistic_model_support = lgs$relative_model_support["Logistic"],
+      clock_outlier = .clock_outlier_stat(u, a),
+      lineage = paste(names(sort(table(ulins), decreasing = TRUE)), collapse = "|"),
+      lineage_summary = lineage_summary,
+      cocirc_lineage_summary = cocirc_summary,
+      region_summary = reg_summary,
+      external_cluster = !(u %in% node_ancestors),
+      tips = paste(tu, collapse = "|"),
+      defining_mutations = paste(cmut$defining, collapse = "|"),
+      all_mutations = paste(cmut$all, collapse = "|"),
+      stringsAsFactors = FALSE
+    )
+
+    for (i in seq_along(test_cluster_odds)) {
+      vn <- test_cluster_odds[i]
+      val <- test_cluster_odds_value[i]
+      clodds <- .var_proportionality_stat(u, ta, var = vn, value = val)
+      X <- cbind(X, t(clodds))
+    }
+    rownames(X) <- as.character(u)
+    if (u %in% report_nodes) { # print progress
+      i <- which(nodes == u)
+      message(paste("Progress", round(100 * i / length(nodes)), "%"))
+    }
+    if (detailed_output) {
+      cldir <- glue::glue("{output_dir}/{as.character(u)}")
```

Comment:
This whole block is now wrapped in if (detailed_output) {}, this was introduced in the mrc-ide repo. The code block hasn't really changed, it's just been right-shifted by two spaces. Again, compare against https://github.com/jumpingrivers/tfpscanner/commit/a08bbd668c7f20eb4865af9cdac1dbd2d583f7b6 to see changes against a styled version of the mrc-ide branch.

Reply:
Checked against this commit and it's just styling changes :)